Sources/NIOPosix/RawSocketBootstrap.swift

```diff
@@ -0,0 +1,197 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import NIOCore
+
+/// A `RawSocketBootstrap` is an easy way to interact with IP based protocols other then TCP and UDP.
+///
+/// Example:
+///
+/// ```swift
+///     let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+///     defer {
+///         try! group.syncShutdownGracefully()
+///     }
+///     let bootstrap = RawSocketBootstrap(group: group)
+///         .channelInitializer { channel in
+///             channel.pipeline.addHandler(MyChannelHandler())
+///         }
+///     let channel = try! bootstrap.bind(host: "127.0.0.1", ipProtocol: .icmp).wait()
+///     /* the Channel is now ready to send/receive IP packets */
+///
+///     try channel.closeFuture.wait()  // Wait until the channel un-binds.
+/// ```
+///
+/// The `Channel` will operate on `AddressedEnvelope<ByteBuffer>` as inbound and outbound messages.
+public final class RawSocketBootstrap {
```

Comment:
```suggestion
public final class NIORawSocketBootstrap {
```

as per the public API guidelines, this need to have a `NIO` prefix

Reply:
Done in https://github.com/apple/swift-nio/pull/2320/commits/981d0b88b242aa1aa5ed14385699dde39c95cfa1